### DISCUSSION TITLE
Xiangqi NNUE training

### DISCUSSION THREAD

**Discussion by ianfab** [fairy-stockfish/variant-nnue-pytorch#13] [General]
Discussion about technical issues during Xiangqi NNUE training, how to solve them, and sharing of links to generated training data.

See https://github.com/ianfab/variant-nnue-pytorch/wiki/%E4%B8%AD%E6%96%87 for documentation, and feel free to extend it if you have additional information.

**Comment by Newera2022** [comment]


**Reply by ianfab** [reply]
Hi. I think you are confusing something here. The concept of "blocks" is used in deep residual CNNs as used in Alpha/LeelaZero style approaches. The NNUE in (Fairy-)Stockfish is a very shallow neural network without anything like blocks, also see the [documentation](https://github.com/ianfab/variant-nnue-pytorch/wiki/Technical-details#variant-nnue-halfkav2-architecture) and this
![image](https://raw.githubusercontent.com/ianfab/variant-nnue-pytorch/master/docs/img/HalfKAv2-45056-256x2P8x2%5B-32-32-1%5Dx8.png).

**Reply by shuangqiao999** [reply]
Can you release a PyTorch running on CPU

**Reply by ianfab** [reply]
Would be nice to have, but likely won't happen.
